CefSharp.BrowserSubprocess.Core/CefAppUnmanagedWrapper.cpp

```diff
@@ -91,64 +91,105 @@ namespace CefSharp
     {
         auto handled = false;
         auto name = message->GetName();
-        if (name == kEvaluateJavascriptRequest)
+        auto argList = message->GetArgumentList();
+
+        auto browserId = argList->GetInt(0);
+        if (browser->GetIdentifier() != browserId)
         {
-            auto argList = message->GetArgumentList();
-            auto browserId = argList->GetInt(0);
-            auto frameId = GetInt64(argList, 1);
-            auto callbackId = GetInt64(argList, 2);
-            auto script = argList->GetString(3);
+            throw gcnew InvalidOperationException(String::Format("Request BrowserId : {0} does not match browser Id : {1}", browserId, browser->GetIdentifier()));
+        }
+        auto browserWrapper = FindBrowserWrapper(browserId, true);
 
-            if(browser->GetIdentifier() != browserId)
-            {
-                throw gcnew InvalidOperationException(String::Format("Request BrowserId : {0} does not match browser Id : {1}", browserId, browser->GetIdentifier()));
-            }
+        if (name == kEvaluateJavascriptRequest || name == kJavascriptCallbackRequest)
+        {
+            CefRefPtr<CefV8Value> result;
+            CefRefPtr<CefV8Exception> exception;
+            CefRefPtr<CefProcessMessage> response;
+            bool success;
+            int64 callbackId;
 
-            auto browserWrapper = FindBrowserWrapper(browserId, true);
-            auto frame = browser->GetFrame(frameId);
-            if (frame.get())
+            if (name == kEvaluateJavascriptRequest)
             {
-                auto context = frame->GetV8Context();
+                auto frameId = GetInt64(argList, 1);
+                callbackId = GetInt64(argList, 2);
+                auto script = argList->GetString(3);
 
-                if (context.get() && context->Enter())
+                auto frame = browser->GetFrame(frameId);
+                if (frame.get())
                 {
-                    try
+                    auto context = frame->GetV8Context();
+                    V8ContextScope scope(context);
+
+                    success = context->Eval(script, result, exception);
+                    response = CefProcessMessage::Create(kEvaluateJavascriptResponse);
```

Comment:
Should always create the message I think, that way if there's an error we can still respond. It looks no response would be sent if a problem occurred.